src/main/java/com/nvidia/spark/rapids/jni/kudo/KudoHostMergeResult.java

```diff
@@ -30,15 +28,16 @@
  */
 public class KudoHostMergeResult implements AutoCloseable {
   private final Schema schema;
-  private final List<ColumnViewInfo> columnInfoList;
+  private final ColumnViewInfo[] columnInfoList;
   private HostMemoryBuffer hostBuf;
 
-  KudoHostMergeResult(Schema schema, HostMemoryBuffer hostBuf, List<ColumnViewInfo> columnInfoList) {
+  KudoHostMergeResult(Schema schema, HostMemoryBuffer hostBuf, ColumnViewInfo[] columnInfoList) {
     requireNonNull(schema, "schema is null");
     requireNonNull(columnInfoList, "columnInfoList is null");
-    ensure(schema.getFlattenedColumnNames().length == columnInfoList.size(), () ->
-        "Column offsets size does not match flattened schema size, column offsets size: " + columnInfoList.size() +
-            ", flattened schema size: " + schema.getFlattenedColumnNames().length);
+    ensure(schema.getFlattenedColumnNames().length == columnInfoList.length, () ->
```

Comment:
nit: Should this be an assert? On large schemas getFlattenedColumnNames may end up being slightly expensive.